src/components/MainPage/Exporters/SpreadsheetExporter.vue

```diff
@@ -61,24 +97,55 @@ export default {
     async submitForm() {
       this.loading = true;
       try {
-        this.saveExporterProperties({ name, properties: this.properties });
+        if (this.isNewSpreadsheet) {
+          await this.createNewSpreadsheet();
+        }
+
         const result = await saveTransactionsToGoogleSheets(
-          this.properties.fileUrl,
+          this.oauth2Client,
+          this.properties.spreadsheetId,
           this.transactions,
         );
+
+        const statusMessage = result.status === 200
+          ? `There were ${result.existTransactions} transactions in the sheet. `
+            + `We were asked to add ${Object.keys(this.transactions).length} transactions, `
+            + `and now there are ${result.updatedTransactions}.`
+          : result.statusText || 'Unknown error';
+
         this.emitStatus(
-          true,
-          `There were ${result.before} transactions, 
-        we uploaded ${result.new} transactions and now there are ${result.combine} transactions.`,
+          result.status === 200,
+          statusMessage,
         );
+
+        this.saveExporterProperties({ name, properties: this.properties });
       } catch (error) {
-        this.$logger.error(error);
+        this.$logger.error(error.message, error);
         this.emitStatus(false, error.message);
       }
       this.loading = false;
```

Comment:
I think this should be in finally block

Reply:
I think the only advantage of the `finally` block is to run even if the `try` block includes a `return` statement.

But for clean code, I think it cleaner to wrap all the code with `try-catch-finally`.